pkg/bundle/internal/truststore/types.go

```diff
@@ -21,6 +21,7 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
+	"math/rand"
```

Comment:
suggestion: Completely separate from James' concern (which may be a blocker). Sorry for the hastily written comment below!

I think if we're going to do import `math/rand` it should be in a separate package entirely, so it's "quarantined". When we work on this file in the future we don't want to assume that `rand` means `crypto/rand`.

I'd create `internal/insecurerand/insecurerand.go`, have it expose `var Rand io.Reader = rand.New(rand.NewSource(1))`.

That was this file would import `insecurerand.Rand` and `math/rand` would be contained to one file only so it couldn't accidentally be used.

That said, `math/rand.Reader` [says](https://pkg.go.dev/math/rand#NewSource) it's not safe for concurrent use:

> Unlike the default [Source](https://pkg.go.dev/math/rand#Source) used by top-level functions, this source is not safe for concurrent use by multiple goroutines.

Might that be an issue if a user has multiple bundles?

Further to that, is a seeded rand actually what we want? We want a deterministic `io.Reader` - could we create a `bytes.Buffer` with say N bytes all set to zero and use a reader on that instead? It looks like the use of `rand` doesn't actually need that many bytes: https://github.com/SSLMate/go-pkcs12/blob/v0.5.0/pkcs12.go#L705

Reply:
Many good points here! Thanks Ash! I will update this PR - even if it's unlikely it will get merged.

Reply:
I have updated the PR now addressing some of your comments.